daemon/cluster/listen_addr.go

```diff
@@ -0,0 +1,163 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+func resolveListenAddr(specifiedAddr string, defaultListenAddr string) (string, error) {
+	specifiedHost, specifiedPort, err := net.SplitHostPort(specifiedAddr)
+	if err != nil {
+		return "", fmt.Errorf("could not parse listen address %s", specifiedAddr)
```

Comment:
@aaronlehmann We have default port adding in daemon side as well. https://github.com/docker/docker/blob/07dd69df8d6c5f4da0059d49e128fef9277fcb5d/daemon/cluster/cluster.go#L1175